### ISSUE TITLE
[interface-name-prefix] <Documentation does not match behavior>

### ISSUE THREAD

**Issue by BurgersMcSlopshot** [typescript-eslint/typescript-eslint#621]
Using the 1.10.2 typescript-eslint plugin, trying to set a value for the rule, "interface-name-prefix" of "always" or "never" (values indicated in the documentation [here](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/interface-name-prefix.md) ) will cause an error to be thrown.

**Repro**
1) add a rule, "@typescript-eslint/interface-name-prefix": ["always"] to an eslint ruleset
2) run eslint

```JSON
{
  "rules": {
    "@typescript-eslint/interface-name-prefix": ["always"]
  }
}
```

```TS
// your repro code case
```

**Expected Result**
Either the documentation to indicate the range of values allowed for the rule or setting the rule to the values indicated in the documentation to work.

**Actual Result**
ESLint throws an error with any value other than off/warn/error. 

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.10.2` |
| `@typescript-eslint/parser`        | `1.10.2` |
| `TypeScript`                       | `3.2.2` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `10.15.0` |
| `npm`                              | `6.4.1` |

**Comment by bradzacher** [comment]
https://eslint.org/docs/user-guide/configuring#configuring-rules

Please read the documentation for configuring eslint rules.
Your config is incorrect.
